geoutils/raster/raster.py

```diff
@@ -2464,15 +2480,29 @@ def icrop(
         if len(bbox) != 4:
             raise ValueError("bbox must be a list or tuple of four integers: (xmin, ymin, xmax, ymax).")
 
-        # Convert pixel coordinates to georeferenced coordinates using the transform
-        bottom_left = self.transform * (bbox[0], self.height - bbox[1])
-        top_right = self.transform * (bbox[2], self.height - bbox[3])
+        # Crop the raster data based on the given bounding box (bbox), limit the range to the borders of the raster
+        crop_img = self.data[..., bbox[0] : min(self.height, bbox[2]), bbox[1] : min(bbox[3], self.width)]
```

Comment:
Ah sorry I didn't notice this line and @adebardo's comment, good catch.
Indeed, this will load the entire array, then subset it. To circumvent this, we need to use `crop()` as before.

Reply:
We should add a test checking if `Raster.is_loaded` is still `False` after calling `icrop()`.